chia/full_node/block_store.py

```diff
@@ -58,7 +58,11 @@ async def create(cls, db_wrapper: DBWrapper):
         await self.db.execute("CREATE INDEX IF NOT EXISTS height on block_records(height)")
 
         await self.db.execute("CREATE INDEX IF NOT EXISTS hh on block_records(header_hash)")
-        await self.db.execute("CREATE INDEX IF NOT EXISTS peak on block_records(is_peak)")
+
+        # this index was initially created and deployed without where condition "is_peak = 1"
+        # to replace the old index with the new one, uncomment the drop
+        # await self.db.execute("DROP INDEX IF EXISTS peak on block_records(is_peak)")
+        await self.db.execute("CREATE INDEX IF NOT EXISTS peak on block_records(is_peak) where is_peak = 1")
```

Comment:
I think the name of this index need to be something other than "peak", since that's the name of the current index. IF they have the same name, the commented-out `DROP INDEX IF EXISTS` would end up dropping both the new and the old index.

Reply:
unless of course we don't have the ambition of upgrading existing users. given that it's potentially expensive to recreate the index, that argument could be made